apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -47,11 +48,17 @@ function tableName(table: string, schema?: string): string{
 /**
  * Do not convert DATE types to JS date.
  * It ignores of applying a wrong timezone to the date.
+ *
+ * See also: https://github.com/brianc/node-postgres/issues/285
+ * (and note that the code refrenced in /lib/textParsers.js has been broken out into it own module
+ * so it now lives in https://github.com/brianc/node-pg-types/blob/master/lib/textParsers.js#L175)
+ *
  * TODO: do not convert as well these same types with array (types 1115, 1182, 1185)
  */
 pg.types.setTypeParser(1082, 'text', (val) => val); // date
 pg.types.setTypeParser(1114, 'text', (val) => val); // timestamp without timezone
 pg.types.setTypeParser(1184, 'text', (val) => val); // timestamp
+// pg.types.setTypeParser(1186, 'text', (val) => val); // interval (Issue #1442 "BUG: INTERVAL columns receive wrong value when cloning row)
```

Comment:
@davidkaufman looks like some testing changes here need to be reverted before I merge it?

Reply:
Oh, duh yes! fixed that :-)

And -- yay! 😁  I simplified the test and finally got it to detect the problem!

~~~patch
dkaufman@bender studio % yarn test:integration --runInBand --verbose postgres -t issue-1442                                                          yarn run v1.22.19
$ TEST_MODE=1 ELECTRON_RUN_AS_NODE=1 yarn electron ../../node_modules/jest/bin/jest.js --config ./jest.integration.config.js --runInBand --verbose postgres -t issue-1442
$ /Users/dkaufman/dev/bks/beekeeper-studio-fork/node_modules/.bin/electron ../../node_modules/jest/bin/jest.js --config ./jest.integration.config.js --runInBand --verbose postgres -t issue-1442

  console.info tests/integration/lib/db/clients/postgres.spec.ts:161
+++ inserted data:  { id: 1, amount_of_time: '00:15:00' }

  console.log tests/integration/lib/db/clients/postgres.spec.ts:168
+++ retrieved data:  { id: 1, amount_of_time: PostgresInterval { minutes: 15 } }

 FAIL  tests/integration/lib/db/clients/postgres.spec.ts (24.422s)
  Postgres [9.4 - socket? false]
    ✕ Should clone interval values in pg-intervalStyle format not json (issue-1442) (32ms)

  ● Postgres [9.4 - socket? false] › Should clone interval values in pg-intervalStyle format not json (issue-1442)

    expect(received).toStrictEqual(expected) // deep equality

- Expected
+ Received

  Object {
-   "amount_of_time": "00:15:00",
+   "amount_of_time": PostgresInterval {
+     "minutes": 15,
+   },
    "id": 1,
  }

      169 |
      170 |       // retrieved interval value should be the same, unchanged interval (string) "00:15:00"
    > 171 |       expect ( retrievedData ).toStrictEqual({
          |                                ^
      172 |         id: 1,
      173 |         amount_of_time: insertedValue // should still be a string not an object
      174 |       })

      at Object.<anonymous> (tests/integration/lib/db/clients/postgres.spec.ts:171:32)
          at runMicrotasks (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 24 skipped, 25 total
Snapshots:   0 total
Time:        25.943s, estimated 52s
Ran all test suites matching /postgres/i with tests matching "issue-1442".
error Command failed with exit code 1.
~~~

Not sure what i was doing wrong before ...probably an insufficient coffee dosage

Also implemented @MasterOdin's suggestion to replace all those obscure datatype id's with the `pg.types.builtins` constants